### PR TITLE
demoRelease_register_product_sync_scheduledTask

### PR DESCRIPTION
**Pull request by MohammadrezaTaghipour**
pull request for #13 ;
ask for code review.

### CODE REVIEW

File: src/Plugin/PayamGostarService/Catalog/IPgProductMetaDataService.cs [hunk 1]
```diff
@@ -11,11 +13,18 @@ public partial interface IPgProductMetaDataService
...
+        /// <param name="Id">PgProduct MetaData identifier</param>
         /// <returns>PgProduct MetaData</returns>
         PgProductMetaData GetPgProductMetaDataById(int Id);
 
         /// <summary>
+        /// Gets PgProduct MetaData
+        /// </summary>
+        /// <param name="Code">PgProduct MetaData identifier</param>
+        /// <returns>PgProduct MetaData</returns>
+        PgProductMetaData GetPgProductMetaDataByCode(int Code);
```
Comment (Borzoo) [inline src/Plugin/PayamGostarService/Catalog/IPgProductMetaDataService.cs]: Method arguments must Use pascal casing

File: src/Plugin/PayamGostarService/Catalog/IPgProductMetaDataService.cs [hunk 2]
```diff
@@ -11,11 +13,18 @@ public partial interface IPgProductMetaDataService
...
+
+        /// <summary>
         /// Deletes a PgProduct MetaData
```
Comment (Borzoo) [inline src/Plugin/PayamGostarService/Catalog/IPgProductMetaDataService.cs]: Typo?

File: src/Plugin/PayamGostarService/Catalog/PgProductMetaDataService.cs [hunk 3]
```diff
@@ -1,40 +1,96 @@
...
         # region Implementation of IPgProductMetaDataService
 
         public virtual PgProductMetaData GetPgProductMetaDataById(int Id)
         {
-            throw new NotImplementedException();
+            if (Id <= 0)
+                return null;
+
+            var query = from p in _pgProductMetaDataRepository.Table
+                        where p.Code == Id
```
Comment (Borzoo) [inline src/Plugin/PayamGostarService/Catalog/PgProductMetaDataService.cs]: P.id?

File: src/Plugin/PayamGostarService/Catalog/PgProductMetaDataService.cs [hunk 4]
```diff
@@ -1,40 +1,96 @@
...
-            throw new NotImplementedException();
+            if (pgProductMetaData == null)
+                throw new ArgumentNullException("pgProductMetaData");
+
+            _pgProductMetaDataRepository.Insert(pgProductMetaData);
+        }
+
+        public virtual void InsertPgProductMetaData(IEnumerable<PgProductMetaData> pgProductMetaDataList)
+        {
+            if (pgProductMetaDataList == null && !pgProductMetaDataList.Any())
```
Comment (Borzoo) [inline src/Plugin/PayamGostarService/Catalog/PgProductMetaDataService.cs]: Or

File: src/Plugin/PayamGostarService/Catalog/PgProductMetaDataService.cs [hunk 5]
```diff
@@ -1,40 +1,96 @@
...
-            throw new NotImplementedException();
+            if (pgProductMetaData == null)
+                throw new ArgumentNullException("pgProductMetaData");
+
+            _pgProductMetaDataRepository.Update(pgProductMetaData);
         }
 
+        public virtual void UpdatePgProductMetaData(IEnumerable<PgProductMetaData> pgProductMetaDataList)
+        {
+            if (pgProductMetaDataList == null && !pgProductMetaDataList.Any())
```
Comment (Borzoo) [inline src/Plugin/PayamGostarService/Catalog/PgProductMetaDataService.cs]: Or

File: src/PgNopIntegration.sln [hunk 6]
```diff
@@ -1,7 +1,7 @@
 ﻿
 Microsoft Visual Studio Solution File, Format Version 12.00
-# Visual Studio 14
-VisualStudioVersion = 14.0.24720.0
+# Visual Studio 2013
+VisualStudioVersion = 12.0.40629.0
```
Comment (Borzoo) [inline src/PgNopIntegration.sln]: Ba 2015 kar koni behtare

File: src/Plugin/Data/PgProductIntegrationObjectContext.cs [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+        {
+            return ((IObjectContextAdapter)this).ObjectContext.CreateDatabaseScript();
+        }
+
+        public new IDbSet<TEntity> Set<TEntity>() where TEntity : BaseEntity
+        {
+            return base.Set<TEntity>();
+        }
+
+        public System.Collections.Generic.IList<TEntity> ExecuteStoredProcedureList<TEntity>(string commandText, params object[] parameters) where TEntity : BaseEntity, new()
```
Comment (Borzoo) [inline src/Plugin/Data/PgProductIntegrationObjectContext.cs]: Khode nop ye base nadare ke in methodaye general ro dashte bashe?

File: src/Plugin/Domain/PgProductMetaData.cs [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+    {
+        /// <summary>
+        /// Gets or sets the nopCommerce product identifier
+        /// </summary>
+        public int ProductId { get; set; }
+        /// <summary>
+        /// Gets or sets the Payam Gostar Product metaData code
+        /// </summary>
+        public int Code { get; set; }
+        /// <summary>
```
Comment (Borzoo) [inline src/Plugin/Domain/PgProductMetaData.cs]: Code is string

File: src/Plugin/PayamGostarService/Tasks/ProductSyncTask.cs [hunk 9]
```diff
@@ -0,0 +1,69 @@
+﻿using System;
+using System.Linq;
+using Nop.Services.Catalog;
+using Nop.Services.Tasks;
+using Septa.PayamGostar.ServiceLayer.ServiceType.Api;
+using Septa.PgNopIntegration.Plugin.PayamGostarService.Catalog;
+
+namespace Septa.PgNopIntegration.Plugin.PayamGostarService.Tasks
+{
+    public partial class ProductSyncTask : ITask
```
Comment (Borzoo) [inline src/Plugin/PayamGostarService/Tasks/ProductSyncTask.cs]: Belongs to another pr for issue #10, exclude from this pr.

File: src/Plugin/Plugin.csproj [hunk 10]
```diff
@@ -222,17 +222,24 @@
...
+    <Compile Include="Domain\PgProductMetaData.cs" />
     <Compile Include="PayamGostarPluginDependencyRegistrar.cs" />
+    <Compile Include="PayamGostarService\Catalog\IPgProductMetaDataService.cs" />
+    <Compile Include="PayamGostarService\Catalog\PgProductMetaDataService.cs" />
     <Compile Include="PayamGostarService\IProductSyncService.cs" />
     <Compile Include="PayamGostarService\Product.cs" />
     <Compile Include="PayamGostarService\ProductChanges.cs" />
     <Compile Include="PayamGostarService\ProductSyncService.cs" />
+    <Compile Include="PayamGostarService\Tasks\ProductSyncTask.cs" />
     <Compile Include="Properties\AssemblyInfo.cs" />
```
Comment (Borzoo) [inline src/Plugin/Plugin.csproj]: Exclude from this pr.

**Comment by Borzoo** [comment]
Include closes #9 in pr message.

File: src/Plugin/PayamGostarService/Catalog/IPgProductMetaDataService.cs (hunk 2 above)
Comment (MohammadrezaTaghipour) [inline src/Plugin/PayamGostarService/Catalog/IPgProductMetaDataService.cs]: yes, we don't need delete!
only insert.

File: src/Plugin/Data/PgProductIntegrationObjectContext.cs (hunk 7 above)
Comment (MohammadrezaTaghipour) [inline src/Plugin/Data/PgProductIntegrationObjectContext.cs]: gashtam! chizi be chshmam nakhord.

File: src/Plugin/PayamGostarService/Catalog/IPgProductMetaDataService.cs (hunk 2 above)
Comment (Borzoo) [inline src/Plugin/PayamGostarService/Catalog/IPgProductMetaDataService.cs]: esme method inserte, documentationesh neveshte delete
